templates/tfengine/components/cicd/main.tf

```diff
@@ -211,7 +210,7 @@ resource "google_project_iam_member" "cloudbuild_sa_project_iam" {
   for_each = toset(local.cloudbuild_devops_roles)
   project  = var.project_id
   role     = each.key
-  member   = local.cloudbuild_sa
+  member   = "serviceAccount:${var.service_account_email}"
```

Comment:
You can keep the `cloudbuild_sa` local variable, and just gave it different values based on different cases:

1. `service_account_email` is specified, then `local.cloudbuild_sa` will be "serviceAccount:${var.service_account_email}".
2. service_account_email is not specified, then `local.cloudbuild_sa` will be the current value.